FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -1395,11 +1395,12 @@ void OpDispatchBuilder::XCHGOp(OpcodeArgs) {
 }
 
 void OpDispatchBuilder::CDQOp(OpcodeArgs) {
-  OrderedNode *Src = LoadSource(GPRClass, Op, Op->Src[0], Op->Flags);
   uint8_t DstSize = GetDstSize(Op);
   uint8_t SrcSize = DstSize >> 1;
+  OrderedNode *Src = LoadGPRRegister(X86State::REG_RAX, SrcSize, 0, true);
```

Comment:
Why did you change this? We typically use LoadSource

Reply:
If you  want to eliminate the masking, what you want is `{.AllowUpperGarbage = true}` which opts into loading garbage values in the upper bits.

Reply:
> If you want to eliminate the masking, what you want is `{.AllowUpperGarbage = true}` which opts into loading garbage values in the upper bits. See #3381 as an example

But `LoadSource` takes `const LoadSourceOptions& Options`, while `LoadGPRRegister` takes a `bool AllowUpperGarbage`.

Reply:
Either route is okay to take. In a lot of cases for hardcoded register usage inside of an instruction we will use `LoadGPRRegister`. Since the cwd family of instructions hardcode their register usage to RAX then I think it is fine to use `LoadGPRRegister`.

If the instruction wasn't hardcoded to a register then `LoadSource` is definitely preferred if possible.